apm-sniffer/apm-sdk-plugin/dbcp-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/dbcp/v2/PoolingSetUrlInterceptor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.skywalking.apm.plugin.dbcp.v2;
+
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.plugin.jdbc.connectionurl.parser.URLParser;
+import org.apache.skywalking.apm.plugin.jdbc.trace.ConnectionInfo;
+
+import java.lang.reflect.Method;
+
+/**
+ * {@link PoolingSetUrlInterceptor} intercepted the method of DBCP set url Get parameters.
+ */
+public class PoolingSetUrlInterceptor implements InstanceMethodsAroundInterceptor {
+    private static final String METER_NAME = "datasource";
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        ConnectionInfo connectionInfo = URLParser.parser((String) allArguments[0]);
+        String tagValue = connectionInfo.getDatabaseName() + "_" + connectionInfo.getDatabasePeer();
+        if (tagValue != null) {
```

Comment:
As I understood, it could not be null?

Reply:
> As I understood, it could not be null?

The external call may be set to null



> As I understood, it could not be null?
change to
    if (allArguments[0] != null) {
            ConnectionInfo connectionInfo = URLParser.parser((String) allArguments[0]);
            String tagValue = connectionInfo.getDatabaseName() + "_" + connectionInfo.getDatabasePeer();
            objInst.setSkyWalkingDynamicField(tagValue);
        }

Reply:
I think hrmeans _ is in the string combining. So, it can't be null.

Reply:
> I think he means _ is in the string combining. So, it can't be null.

Exactly.